openstf/pipeline/optimize_hyperparameters.py

```diff
@@ -68,11 +117,44 @@ def optimize_hyperparameters_pipeline(
     study = optuna.create_study(
         pruner=optuna.pruners.MedianPruner(n_warmup_steps=5), direction="minimize"
     )
+    model_type = pj["model"]
+    model = ModelCreator.create_model(model_type)
+
+    # Combine the default parameters with all model specific parameters
+    paramspace = default_paramspace.copy()
+    paramspace.update(**xgb_paramspace, **lgb_paramspace)
+
+    model_params = get_relevant_model_paramspace(model, paramspace)
+
+    if model_type == "lgb":
+
+        model_params[
+            "objective"
+        ] = objective.eval_metric  # The objective of lgb is the eval metric
+        pruning_function = optuna.integration.LightGBMPruningCallback
+        args_eval = {"metric": objective.eval_metric, "valid_name": "valid_1"}
+        if objective.eval_metric == "mae":
+
+            args_eval["metric"] = "l1"
+    else:
+        # for other models use the default objective by not setting it.
+        pruning_function = optuna.integration.XGBoostPruningCallback
+        args_eval = {"observation_key": "validation_1-{}".format(objective.eval_metric)}
```

Comment:
let's try and see if we can find a better place to incorporate this distinction. 
Perhaps the 'get_relevant_model_paramspace' can handle this logic, since it already receives the model as an input?

Reply:
I would suggest to have the optimization model specific stuff in only one place and keep it out of the pipeline. The pipeline can just use the factories to get a model and objective and run the optuna optimization and does not have to be responsible for the model specific stuff.

```shell
openstf.model.objectives -> hold the objectives for the various implementations (xgb.py, lgbm.py etc.)
  can be one file for now if we want instead of a package ofc.

# Or
openstf.model.regressors.objectives > hold the objectives for the various implementations (xgb.py, lgbm.py etc.)
  can be one file for now if we want instead of a package ofc.
# Or 
openstf.model.regressors.<model-name> -> model.py and objective.py

```